python/tests.py

```diff
@@ -0,0 +1,69 @@
+import nix
+import unittest
+
+
+class TestPythonNix(unittest.TestCase):
+    def test_dict(self):
+        val = dict(a=1)
+        self.assertEqual(nix.callExprString("{ a }: a", arg=dict(a=val)), val)
```

Comment:
As [previously hinted](https://github.com/NixOS/nix/pull/7735#discussion_r1129621459), in the [latest commit](https://github.com/NixOS/nix/pull/7735/commits/c87a1b44ff6a05853edf4f202f3ace3d58b06a53) I now changed the API to a more explicit function call like
```python
nix.eval("{ a }: a", arg=dict(a=1))
```
from the original
```python
nix.eval("a", vars=dict(a=1))
```
because that worked the same as `builtins.scopedImport`, which is known as a wart in the Nix language because it doesn't do file caching as `import` does.

However, after noticing that it's now very odd when you don't need to take any arguments, like this:
```python
nix.callExprString("{}: null", arg={})
```
I decided to reinvestigate the original idea, and whether `builtins.scopedImport` is really as bad as we make it out to be, I wrote this down here: https://github.com/NixOS/nix/issues/8024

Looking into this I'm concluding that the three issues listed there shouldn't be a problem for the Python bindings, see the above issue for more context:
- Confusing free variables: Compared to `scopedImport`, the original API allows having the expression string right next to the variables it uses, making it very similar to a function declaration.
- Doesn't allow file evaluation to be cached: The original API can be seen as implicitly prepending the expression with function arguments, and anything under a function argument in a file can't be cached anyways.
- Combined with the ability to shadow builtins: Indeed a problem, but it's neither up to `scopedImport` nor the Python bindings to fix.

Because of that I'm considering going back to the original API

Reply:
Let's not overthink this one. We can replace this function later and focus on other aspects such as a stable intermediate C/C++ interface, validating the packaging (including in Nixpkgs).

> Doesn't allow file evaluation to be cached

You could return a `Value` to cache it from Python. I'll shut up now.

Reply:
I've decided to spend a small amount of time to explore how a custom type can be added, and it doesn't seem so bad.
This sets up a `nix.Value` type
 - https://github.com/tweag/nix/pull/17

It's a bit MVP, but it establishes the pattern.
It has also given me a better insight into how the bindings work.